File: openaev-api/src/main/java/io/openaev/service/connectors/PlatformConnectors.java
package io.openaev.service.connectors;

import io.openaev.injectors.openaev.OpenAEVImplantContract;
import io.openaev.integration.impl.executors.openaev.OpenAEVExecutorIntegration;

/**
 * The two connectors that ARE the platform.
 *
 * <p>The implant injector executes every payload and the agent executor drives every agent, so
 * removing either one breaks execution itself. Every other connector is removable, including the
 * ones running in-process: they are shipped with the platform, not vital to it, and a legacy row
 * whose implementation was dropped from the code in an earlier version has to be cleanable.
 * In-process connectors that are still implemented register again on the next startup.
 *
 * <p>Matched on the connector type rather than the seeded id: ids are per tenant, types are stable.
 */
public final class PlatformConnectors {

  public static boolean isPlatformInjector(String injectorType) {
    return OpenAEVImplantContract.TYPE.equals(injectorType);
  }

  public static boolean isPlatformExecutor(String executorType) {
    return OpenAEVExecutorIntegration.OPENAEV_EXECUTOR_TYPE.equals(executorType);
  }

  private PlatformConnectors() {}
}

File: openaev-front/src/admin/components/integrations/common/platform-connectors.ts
/**
 * Connectors that ARE the platform: the implant injector executes every payload and the agent
 * executor drives every agent, so removing either one breaks execution itself and no amount of
 * re-registration recovers the injects that referenced them.
 *
 * <p>Every other connector is removable, including the ones running in-process: they are shipped
 * with the platform, not vital to it, and a legacy row whose implementation no longer exists (an
 * injector dropped from the code in an earlier version) must be cleanable from the UI. In-process
 * connectors that are still implemented simply register again on the next platform restart.
 *
 * <p>Kept as connector types, not ids: the ids are seeded per tenant, the types are stable.
 */
const PLATFORM_CONNECTOR_TYPES = ['openaev_implant', 'openaev_agent'];

const isPlatformConnector = (connectorType?: string): boolean =>
  connectorType != null && PLATFORM_CONNECTOR_TYPES.includes(connectorType);

export default isPlatformConnector;
